pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TopicStats.java

```diff
@@ -0,0 +1,79 @@
+package org.apache.pulsar.broker.stats.prometheus;
+
+import org.apache.bookkeeper.mledger.impl.ManagedLedgerMBeanImpl;
+import org.apache.bookkeeper.mledger.util.StatsBuckets;
+import org.apache.pulsar.utils.SimpleTextOutputStream;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class TopicStats {
+
+    private static final String TOPIC_REGEX = ".*://";
+
+    int subscriptionsCount;
+    int producersCount;
+    int consumersCount;
+    double rateIn;
+    double rateOut;
+    double throughputIn;
+    double throughputOut;
+
+    long storageSize;
+    public long msgBacklog;
+
+    StatsBuckets storageWriteLatencyBuckets = new StatsBuckets(ManagedLedgerMBeanImpl.ENTRY_LATENCY_BUCKETS_USEC);
+    StatsBuckets entrySizeBuckets = new StatsBuckets(ManagedLedgerMBeanImpl.ENTRY_SIZE_BUCKETS_BYTES);
+    double storageWriteRate;
+    double storageReadRate;
+
+    Map<String, AggregatedReplicationStats> replicationStats = new HashMap<>();
+
+    public void reset() {
+        subscriptionsCount = 0;
+        producersCount = 0;
+        consumersCount = 0;
+        rateIn = 0;
+        rateOut = 0;
+        throughputIn = 0;
+        throughputOut = 0;
+
+        storageSize = 0;
+        msgBacklog = 0;
+        storageWriteRate = 0;
+        storageReadRate = 0;
+
+        replicationStats.clear();
+        storageWriteLatencyBuckets.reset();
+        entrySizeBuckets.reset();
+    }
+
+    static void printNamespaceStats(SimpleTextOutputStream stream, String cluster, String namespace, String topic,
+        TopicStats stats) {
+
+        topic = cleanTopicName(topic);
+
```

Comment:
@merlimat are there more metrics we should export for topics? Should we keep the topic-type:// in the name when exporting?